### PR TITLE
Port cross-compile tests for java.{io|net}.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Will need a rebase on #2043.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+class URITest {
+
+  def expectURI(uri: URI, isAbsolute: Boolean, isOpaque: Boolean)(
+    authority: String = null, fragment: String = null,
+    host: String = null, path: String = null, port: Int = -1,
+    query: String = null, scheme: String = null, userInfo: String = null,
+    schemeSpecificPart: String = null)(rawAuthority: String = authority,
+    rawFragment: String = fragment, rawPath: String = path,
+    rawQuery: String = query, rawUserInfo: String = userInfo,
+    rawSchemeSpecificPart: String = schemeSpecificPart): Unit = {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala]: 4 spaces

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+    }
+    assertEquals(rawQuery, uri.getRawQuery())
+    assertEquals(rawSchemeSpecificPart, uri.getRawSchemeSpecificPart())
+    assertEquals(rawUserInfo, uri.getRawUserInfo())
+    assertEquals(scheme, uri.getScheme())
+    assertEquals(schemeSpecificPart, uri.getSchemeSpecificPart())
+    assertEquals(userInfo, uri.getUserInfo())
+    assertEquals(isAbsolute, uri.isAbsolute())
+    assertEquals(isOpaque, uri.isOpaque())
+
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala]: Spurious blank line.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.scalajs.testsuite.util
+
+object Platform {
+  /** Returns `true` if and only if the code is executing on a JVM.
+    *  Note: Returns `false` when executing on any JS VM.
+    */
+  lazy val executingInJVM: Boolean = {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala]: This should not be there. There was something for that in the math PR.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/ByteArrayInputStreamTest.scala [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+\*                                                                      */
+package org.scalajs.testsuite.javalib.io
+
+import java.io._
+
+import scala.language.implicitConversions
+
+class ByteArrayInputStreamTest extends CommonStreamsTests {
+  byteArrayInputStreamLikeTests(seq =>
+    new ByteArrayInputStream(seq.map(_.toByte).toArray))
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/io/ByteArrayInputStreamTest.scala]:
```scala
  byteArrayInputStreamLikeTests { seq =>
    new Byte...
  }
```

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+  }
+
+  @Test def `should_parse_vanilla_absolute_URIs`(): Unit = {
+    expectURI(new URI("http://java.sun.com/j2se/1.3/"), true, false)(
+      scheme = "http",
+      host = "java.sun.com",
+      path = "/j2se/1.3/",
+      authority = "java.sun.com",
+      schemeSpecificPart = "//java.sun.com/j2se/1.3/")()
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/net/URITest.scala]: 4 spaces (a lot of those in this file)

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala (hunk 3 above)
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/util/Platform.scala]: I think you forgot to remove this file.

**Comment by sjrd** [comment]
LGTM

**Pull request merged**